File: Easy/Python/2591. Distribute Money to Maximum Children.py
class Solution(object):
    def distMoney(self, money, children):
        if money < children:
            return -1
        money -= children

        count = 0
        while 7 <= money and count != children:
            money -= 7
            count += 1
        
        if money != 0 and ((count == children) or (money == 3 and children - count == 1)):
            count -= 1
        
        return count

File: Easy/JavaScript/2591. Distribute Money to Maximum Children.js
var distMoney = function(money, children) {
    if(money < children)
        return -1;
    money -= children;

    let count = 0;
    while(7 <= money && count != children){
        money -= 7;
        count++;
    }

    if(money != 0 && ((count == children) || (money == 3 && children - count == 1)))
        count--;
    
    return count;
};
